tools/SourceKit/tools/sourcekitd/bin/InProc/sourcekitdInProc.cpp

```diff
@@ -82,14 +83,17 @@ UIdent sourcekitd::UIdentFromSKDUID(sourcekitd_uid_t uid) {
 std::string sourcekitd::getRuntimeLibPath() {
 #if defined(_WIN32)
   MEMORY_BASIC_INFORMATION mbi;
-  char path[MAX_PATH + 1];
+  llvm::SmallString<MAX_PATH + 1> path;
+  path.assign(MAX_PATH + 1, '\0');
   if (!VirtualQuery(static_cast<void *>(sourcekitd_initialize), &mbi,
                     sizeof(mbi)))
     llvm_unreachable("call to VirtualQuery failed");
-  if (!GetModuleFileNameA(static_cast<HINSTANCE>(mbi.AllocationBase), path,
+  if (!GetModuleFileNameA(static_cast<HINSTANCE>(mbi.AllocationBase), path.begin(),
                           MAX_PATH))
     llvm_unreachable("call to GetModuleFileNameA failed");
-  return llvm::sys::path::parent_path(path);
+  path = llvm::sys::path::parent_path(llvm::sys::path::parent_path(path));
```

Comment:
You may be getting into undefined behavior here, initializing `path` from its own memory.
Could you follow similar method to how libclang does this here: https://github.com/apple/swift-clang/blob/078517eab71c45f8a3fc066c02b642bda9c87422/tools/libclang/CIndexer.cpp#L49

Reply:
Yeah probably.. fixed now!